lib/admin/WP_Auth0_Admin_Appearance.php

```diff
@@ -53,117 +53,91 @@ public function init() {
-	public function render_primary_color() {
-    $v = $this->options->get( 'primary_color' );
-?>
-      <input type="text" name="<?php echo $this->options->get_options_name(); ?>[primary_color]" id="wpa0_primary_color" value="<?php echo esc_attr( $v ); ?>" />
-      <div class="subelement">
-        <span class="description"><?php echo __( 'The primary color for Lock', 'wp-auth0' ); ?></span>
-      </div>
-    <?php
-  }
-
-  public function render_language_dictionary() {
-    $v = $this->options->get( 'language_dictionary' );
-?>
-      <textarea name="<?php echo $this->options->get_options_name(); ?>[language_dictionary]" id="wpa0_language_dictionary"><?php echo esc_attr( $v ); ?></textarea>
-      <div class="subelement">
-        <span class="description"><?php echo __( 'This is the widget\'s languageDictionary param.', 'wp-auth0' ); ?><a target="_blank" href="https://github.com/auth0/lock#ui-options"><?php echo __( 'More info', 'wp-auth0' ); ?></a></span>
-      </div>
-    <?php
-  }
-
-	public function render_custom_css() {
-		$v = $this->options->get( 'custom_css' );
-?>
-      <textarea name="<?php echo $this->options->get_options_name(); ?>[custom_css]" id="wpa0_custom_css"><?php echo esc_attr( $v ); ?></textarea>
-      <div class="subelement">
-        <span class="description"><?php echo __( 'This should be a valid CSS to customize the Auth0 login widget. ', 'wp-auth0' ); ?><a target="_blank" href="https://github.com/auth0/wp-auth0#can-i-customize-the-login-widget"><?php echo __( 'More info', 'wp-auth0' ); ?></a></span>
-      </div>
-    <?php
+	public function render_language( $args = array() ) {
+		$this->render_text_field( $args[ 'label_for' ], $args[ 'opt_name' ] );
+		$this->render_field_description(
+			__( 'The language parameter for the Auth0 login form. ', 'wp-auth0' ) .
+			sprintf(
+				'<a href="https://github.com/auth0/lock/tree/master/src/i18n" target="_blank">%s</a>',
+				__( 'Available languages list', 'wp-auth0' )
+			)
+		);
 	}
 
-	public function render_custom_js() {
-		$v = $this->options->get( 'custom_js' );
-?>
-      <textarea name="<?php echo $this->options->get_options_name(); ?>[custom_js]" id="wpa0_custom_js"><?php echo esc_attr( $v ); ?></textarea>
-      <div class="subelement">
-        <span class="description"><?php echo __( 'This should be a valid JS to customize the Auth0 login widget to, for example, add custom buttons. ', 'wp-auth0' ); ?><a target="_blank" href="https://auth0.com/docs/hrd#option-3-adding-custom-buttons-to-lock"><?php echo __( 'More info', 'wp-auth0' ); ?></a></span>
-      </div>
-    <?php
+	public function render_primary_color( $args = array() ) {
+		$this->render_text_field( $args[ 'label_for' ], $args[ 'opt_name' ] );
+		$this->render_field_description(
+			__( 'Primary color for the Auth0 login form in hex format. ', 'wp-auth0' ) .
+			$this->get_docs_link(
+				'libraries/lock/v11/configuration#primarycolor-string-',
+				__( 'More information this settings', 'wp-auth0'	)
```

Comment:
> More information this settings
????

Reply:
It's like you read these things over and over and still ...